### PR TITLE
feat(snowflake): enhance pushdown_deny_usernames by allowing patterns and add pushdown_allow_usernames

### PR DESCRIPTION
**Pull request by sgomezvillamor**
## Summary

This PR adds a new `pushdown_allow_usernames` configuration field to the Snowflake connector for improved performance by filtering in only specific users for lineage/usage/queries extraction. Additionally, it enhances the existing `pushdown_deny_usernames` functionality with pattern support and improves security.

### 🆕 New Feature: `pushdown_allow_usernames`
- **New config field**: `pushdown_allow_usernames: List[str]` (defaults to empty list)
- **Pattern support**: Accepts SQL LIKE patterns (e.g., `'ANALYST_%'`, `'%_USER'`, `'MAIN_ACCOUNT'`)
- **Use case**: Filter in only specific users to improve query performance on large Snowflake deployments
- **Logic**: When specified, only users matching the allow patterns (and not in deny list) are included

### 🔧 Enhanced: `pushdown_deny_usernames` 
- **Pattern support**: Now supports SQL LIKE patterns in addition to exact matches (non-breaking change)
- **Updated documentation**: Clarified that patterns are supported with examples
- **Backward compatible**: Existing exact match configurations continue to work unchanged

### Other Improvements
- **SQL injection protection**: Both deny and allow patterns now properly escape single quotes (`'` → `''`)
- **Refactored filtering logic**: Extracted `_build_user_filter` method for better testability and maintainability

### Configuration Flow
```
SnowflakeV2Config → SnowflakeQueriesExtractorConfig → SnowflakeQueriesExtractor → QueryLogQueryBuilder
```

### SQL Generation Logic
- **Deny patterns**: `user_name NOT ILIKE 'pattern'` (combined with AND)
- **Allow patterns**: `user_name ILIKE 'pattern'` (combined with OR)  
- **Combined**: `(deny_conditions) AND (allow_conditions)`
- **Case-insensitive**: Uses `ILIKE` for pattern matching

### Example Usage
```yaml
source:
  type: snowflake
  config:
    # Exclude service accounts and admin users
    pushdown_deny_usernames:
      - "SERVICE_%"
      - "ADMIN_%"
      - "SYSTEM_USER"
    
    # Include only analysts and specific users  
    pushdown_allow_usernames:
      - "ANALYST_%"
      - "%_USER"
      - "MAIN_ACCOUNT"
```

## Testing

- **17 comprehensive test cases** covering all combinations of deny/allow patterns
- **Edge cases**: Empty lists, null values, overlapping patterns, special characters
- **Security tests**: Validate SQL injection protection with malicious inputs
- **All existing tests pass**: No regressions introduced

## Non-Breaking Changes

✅ **`pushdown_deny_usernames` pattern support is backward compatible**:
- Existing configs: `['SERVICE_USER', 'ADMIN_USER']` → Still works exactly the same
- New pattern configs: `['SERVICE_%', 'ADMIN_%']` → Now supported  
- Exact matches are simply a subset of pattern matching

✅ **New `pushdown_allow_usernames` field**:
- Defaults to empty list (no filtering)
- Only affects behavior when explicitly configured
- No impact on existing deployments

## Performance Impact

- **Positive**: Reduces query volume by filtering users at the SQL level
- **Use case**: Particularly beneficial for large Snowflake deployments with high-volume service accounts
- **Efficient**: Uses native Snowflake `ILIKE` for optimal performance

### CODE REVIEW

File: metadata-ingestion/tests/unit/snowflake/test_snowflake_queries.py [hunk 1]
```diff
@@ -229,3 +226,121 @@ def test_query_with_combined_database_filtering(self):
 
         # SQL parsing should succeed
         sqlglot.parse(query, dialect=Snowflake)
+
+
+class TestBuildUserFilter:
+    @pytest.mark.parametrize(
+        "deny_usernames,allow_usernames,expected",
```
Comment (treff7es) [inline metadata-ingestion/tests/unit/snowflake/test_snowflake_queries.py:233]: I think this can be simplified and a bit more readable by:
```
  @pytest.mark.parametrize(
      "deny_usernames,allow_usernames,expected",
      [
          (None, None, "TRUE"),
          ([], [], "TRUE"),
          (["SERVICE_USER"], None, "(user_name NOT ILIKE 'SERVICE_USER')"),
          (["SERVICE_%"], None, "(user_name NOT ILIKE 'SERVICE_%')"),
          # ... more simple tuples
      ],
      ids=[
          "no_filters",
          "empty_lists", 
          "single_deny_exact",
          "single_deny_pattern",
          # ... corresponding ids
      ]
  )
```

**Comment by sgomezvillamor** [comment]
CI errors are unrelated - flaky python lineage sdk errors
